src/wrappers/OsipiBase.py

```diff
@@ -0,0 +1,210 @@
+        required_bvalues (int):
+            The lowest number of b-values required.
+        required_thresholds (array-like)
+            1D array-like of two ints [least number of b-value thresholds, most number of b-value_thresholds].
+        required_bounds (bool):
+            Whether bounds are required or not.
+        required_bounds_optional (bool):
+            Whether bounds are optional or not
+        required_initial_guess (bool):
+            Whether an initial guess is required or not.
+        required_initial_guess_optional (bool):
+            Whether an initial guess is optional or not.
+        """
+        print("\n### Algorithm requirements ###")
+        
+        # Check if the attributes exist and print them
+        if hasattr(self, "required_bvalues"):
+            print(f"Number of b-values: {self.required_bvalues}")
+            
+        if hasattr(self, "required_thresholds"):
+            print(f"Numer of b-value thresholds [at least, at most]: {self.required_thresholds}")
+            
+        if hasattr(self, "required_bounds") and hasattr(self, "required_bounds_optional"):
+            if self.required_bounds:
+                print(f"Bounds required: {self.required_bounds}")
+            else:
+                # Bounds may not be required but can be optional
+                if self.required_bounds_optional:
+                    print(f"Bounds required: {self.required_bounds} but is optional")
+                else:
+                    print(f"Bounds required: {self.required_bounds} and is not optional")
+                    
+        if hasattr(self, "required_initial_guess") and hasattr(self, "required_initial_guess_optional"):
+            if self.required_initial_guess:
+                print(f"Initial guess required: {self.required_initial_guess}")
+            else:
+                # Bounds may not be required but can be optional
+                if self.required_initial_guess_optional:
+                    print(f"Initial guess required: {self.required_initial_guess} but is optional")
+                else:
+                    print(f"Initial guess required: {self.required_initial_guess} and is not optional")
+
+    def osipi_accepted_dimensions(self):
+        """The array of accepted dimensions
+        e.g.
+        (1D,   2D,   3D,    4D,    5D,    6D)
+        (True, True, False, False, False, False)
+        """
+        
+        return (False,) * 6
+
+    def osipi_accepts_dimension(self, dim):
+        """Query if the selection dimension is fittable"""
+        
+        accepted = self.accepted_dimensions()
+        if dim < 0 or dim > len(accepted):
+            return False
+        return accepted[dim]
+    
+    def osipi_check_required_bvalues(self):
+        """Checks if the input bvalues fulfil the algorithm requirements"""
+        
+        if self.bvalues.size < self.required_bvalues:
+            print("Conformance error: Number of b-values.")
```

Comment:
How do you see this working? Right now it's a print and a return value. But I also could see it asserting that the sizes are accurate. I suppose it depends how strict we want to be.